imap_processing/ena_maps/utils/spatial_utils.py

```diff
@@ -0,0 +1,270 @@
+    el_bin_edges = np.arange(-90, 90 + el_spacing, el_spacing)
+    el_bin_midpoints = el_bin_edges[:-1] + el_spacing / 2  # Midpoints between edges
+
+    return az_bin_edges, el_bin_edges, az_bin_midpoints, el_bin_midpoints
+
+
+def build_solid_angle_map(
+    spacing: float, input_degrees: bool = True, output_degrees: bool = False
+) -> NDArray:
+    """
+    Build a solid angle map for a given spacing in degrees.
+
+    Parameters
+    ----------
+    spacing : float
+        The bin spacing in the specified units.
+    input_degrees : bool, optional
+        If True, the input spacing is in degrees
+        (default is True for radians).
+    output_degrees : bool, optional
+        If True, the output solid angle map is in square degrees
+        (default is False for steradians).
+
+    Returns
+    -------
+    solid_angle_grid : np.ndarray
+        The solid angle map grid in steradians (default) or square degrees.
+        First index is latitude/el, second index is longitude/az.
+    """
+    if input_degrees:
+        spacing = np.deg2rad(spacing)
+
+    if spacing <= 0:
+        raise ValueError("Spacing must be positive valued, non-zero.")
+
+    if not np.isclose((np.pi / spacing) % 1, 0):
+        raise ValueError("Spacing must divide evenly into pi radians.")
+
+    latitudes = np.arange(-np.pi / 2, np.pi / 2 + spacing, step=spacing)
+    sine_latitudes = np.sin(latitudes)
+    delta_sine_latitudes = np.diff(sine_latitudes)
+    solid_angle_by_latitude = np.abs(spacing * delta_sine_latitudes)
+
+    # Order ensures agreement with build_az_el_grid's order of tiling az/el grid.
+    solid_angle_grid = np.repeat(
+        solid_angle_by_latitude[np.newaxis, :], (2 * np.pi) / spacing, axis=0
+    )
+
+    if output_degrees:
+        solid_angle_grid *= (180 / np.pi) ** 2
+
+    return solid_angle_grid
+
+
+@typing.no_type_check
+def rewrap_even_spaced_el_az_grid(
+    raveled_values: NDArray,
+    shape: tuple[int] | None = None,
+    extra_axis: bool = False,
+    order: typing.Literal["C"] | typing.Literal["F"] = "F",
+) -> NDArray:
+    """
+    Take an unwrapped (raveled) 1D array and reshapes it into a 2D el/az grid.
+
+    Assumes the following must be true of the original grid:
+    1. Grid was evenly spaced in angular space,
+    2. Grid had the same spacing in both azimuth and elevation.
+    3. Elevation is the 0th axis (and extends a total of 180 degrees),
+    4. Azimuth is the 1st axis (and extends a total of 360 degrees).
+
+    Parameters
+    ----------
+    raveled_values : NDArray
+        1D array of values to be reshaped into a 2D grid.
+    shape : tuple[int], optional
+        The shape of the original grid, if known, by default None.
+        If None, the shape will be inferred from the size of the input array.
+    extra_axis : bool, optional
+        If True, input is a 2D array with latter axis being 'extra', non-spatial axis.
+        This axis (e.g. energy bins) will be preserved in the reshaped grid.
```

Comment:
Remove this and just say that the first dimension is always the raveled dimension and anything after that gets preserved? i.e. do this automatically without needing a keyword argument.

Reply:
Ah, thanks for the code suggestion below - that helped me understand how to do this simply.